Define the GitOps Principles (#48)
gitops-working-group/gitops-working-group: GLOSSARY.md

# Glossary

> - **Desired state**

Reviewer: ```suggestion
- **Desired State**
```

>     The aggregate of all configuration data for a system form its "Desired State". The "Desired State" of a system is defined as data sufficient to recreate the system from nothing so that instances of the system are behaviorally indinstinguishable.

Reviewer: I find ".. from nothing.." too vague, although I don't have a perfect better wording. "from scratch" is probably not much better. Maybe leave this out, and write:

```suggestion
    The aggregate of all configuration data for a system form its "Desired State".  The "Desired State" of a system is defined as data sufficient to recreate the system so that instances of the system are behaviourally indistinguishable.
```

- **State Store**
    

>     A system for storing versioned, immutable Desired States that provides access control and auditing on the chnages to the Desired State. Git may be configured as a State Store, but special precautions must be taken.

Reviewer: ```suggestion
    A system for storing versioned, immutable Desired States that provides access control and auditing on the changes to the Desired State. Git may be configured as a State Store, but special precautions must be taken.
```

Maybe link to what "Git may be configured as a State Store, but special precautions must be taken." means exactly?

- **Reconciliation**
  

>   A continuously running attempt to reconcile _Definition_ of desired state with the current state.

Reviewer 2: ```The process by which the current state of a system is compared against and made consistent with the system's desired state as declared in the state store```

I think the concept of reconciliation doesn't necessarily mean it's continuous, rather a one off process that looks to achieve consistency between two different states. It can be made continuous, but that is not implicit.

- **Software System**

>     One or more Runtime environments consisting of resources under management

Reviewer: ```suggestion
    One or more Runtime environments consisting of resources under management.
```

    In each Runtime, management Agents to act on resources according to security policies.
    One or more software Repositories for storing deployable artifacts that may be loaded into the runtime environments, eg. configuration files, code, binaries and packages. 
    One or more Administrators who are responsible for operating the runtime environments ie. installing, starting, stopping and updating software, code, configuration, etc.

>     A set of policies controlling access and management of repos, deployment, runtimes.

Reviewer: ```suggestion
    A set of policies controlling access and management of repositories, deployments, runtimes.
```
